src/app/beer_garden/api/stomp/server.py

```diff
@@ -0,0 +1,136 @@
+import stomp
+import logging
+import time
+import beer_garden.config as config
+from brewtils.schema_parser import SchemaParser
+import beer_garden.events
+import beer_garden.router
+from beer_garden.api.stomp.processors import append_headers, process_send_message
+
+conn = None
+bg_active = False
+
+
+def send_message(message, headers=None):
+    if headers is None:
+        headers = {}
+    global conn
+    stomp_config = config.get("entry.stomp")
+    message, response_headers = process_send_message(message)
+    response_headers = append_headers(
+        response_headers=response_headers, request_headers=headers
+    )
+    if conn.is_connected():
+        if "reply-to" in headers:
+            conn.send(
+                body=message, headers=response_headers, destination=headers["reply-to"]
+            )
+        else:
+            conn.send(
+                body=message,
+                headers=response_headers,
+                destination=stomp_config.event_destination,
+            )
+
+
+def send_error_msg(error_msg, headers):
+    global conn
+    stomp_config = config.get("entry.stomp")
+    error_headers = None
+    error_headers = append_headers(error_headers, headers)
+
+    if conn.is_connected():
+        if "reply-to" in headers:
+            conn.send(
+                body=error_msg,
+                headers=error_headers,
+                destination=headers["reply-to"],
+            )
+        else:
+            conn.send(
+                body=error_msg,
+                headers=error_headers,
+                destination=stomp_config.event_destination,
+            )
+    pass
+
+
+class OperationListener(stomp.ConnectionListener):
+    def on_error(self, headers, message):
+        logger = logging.getLogger(__name__)
+        logger.warning("received an error:", headers)
+
+    def on_message(self, headers, message):
+        try:
+            operation = SchemaParser.parse_operation(message, from_string=True)
+            if hasattr(operation, "kwargs"):
+                operation.kwargs.pop("wait_timeout", None)
+            result = beer_garden.router.route(operation)
+            send_message(result, headers)
+        except Exception as e:
+            send_error_msg(str(e), headers)
+
+
+class Connection:
+    @staticmethod
+    def __init__():
+        global bg_active, conn
+        stomp_config = config.get("entry.stomp")
+        host_and_ports = [(stomp_config.host, stomp_config.port)]
+        bg_active = True
+        conn = stomp.Connection(host_and_ports=host_and_ports, heartbeats=(10000, 0))
+        if stomp_config.use_ssl:
+            conn.set_ssl(
+                for_hosts=host_and_ports,
+                key_file=stomp_config.private_key,
+                cert_file=stomp_config.cert_file,
+            )
+        conn.set_listener("", OperationListener())
+
+    @staticmethod
+    def connect(connected_message=None):
+        global conn
```

Comment:
`global` statements are only needed if you're *assigning* to that variable inside of the function. If you're just accessing / using the value they're not needed.

It's because without them if you tried to assign something to the name `conn` inside the function Python will create a normal function-scoped name for you. The `global` tells Python to not do that and instead assign to the "global" namespace.

If you're just accessing the name you don't have to worry about this because the normal name lookup will look for the name in the function scope, fail to find it there, look in the module scope, and find it.